src/cai_lab_to_nwb/zaki_2024/interfaces/zaki_2024_shock_stimuli_interface.py

```diff
@@ -0,0 +1,50 @@
+"""Primary class for converting experiment-specific behavior."""
+
+from pynwb.file import NWBFile
+
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import DeepDict
+from typing import Optional, List
+from pynwb.epoch import TimeIntervals
+
+
+class Zaki2024ShockStimuliInterface(BaseDataInterface):
+    """Adds annotated events of shock times."""
+
+    keywords = ["behavior", "sleep stages"]
+
+    def __init__(self, verbose: bool = False):
+
+        self.verbose = verbose
+        super().__init__()
+
+    def get_metadata(self) -> DeepDict:
+        # Automatically retrieve as much metadata as possible from the source files available
+        metadata = super().get_metadata()
+
+        return metadata
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        shock_amplitude: float,
+        shock_times: List = [120.0, 180.0, 240.0],
+        shock_duration: float = 2.0,
+        metadata: Optional[dict] = None,
+    ):
+
+        description = (
+            "During aversive encoding, after a baseline period of 2 min, mice received three 2 s foot shocks "
+            "of either amplitude 0.25 mA (low-shock) or 1.5 mA (high-shock), with an intershock interval of 1 min. "
+            "All testing was done in Med Associates chambers. "
+        )
+
+        shock_stimuli = TimeIntervals(name="ShockStimuli", description=description)
+        column_description = "Shock amplitude in mA"
+        shock_stimuli.add_column(name="shock_amplitude", description=column_description)
+        for start_time in shock_times:
+            shock_stimuli.add_interval(
+                start_time=start_time, stop_time=start_time + shock_duration, shock_amplitude=shock_amplitude
+            )
+
+        nwbfile.add_time_intervals(shock_stimuli)
```

Comment:
Maybe they should be added to stimuli?

Reply:
I was not sure if stimuli referred only to current injection in patch clamp exp or similar. @weiglszonja, what do you usually do when you have external stimuli as shocks?

Reply:
OK, I say merge this and then we can change that to stimuli if that comes from the discussion. I am not even sure a TimeIntervals can be written in one of the stimuli containers (template, stimuli)

Reply:
Hmm, looking up the schema for `stimulus`:

> 'Data pushed into the system (eg, video stimulus, sound, voltage, etc) and
>       secondary representations of that data (eg, measurements of something used as
>       a stimulus). This group should be made read-only after experiment complete and timestamps
>       are corrected to common timebase. Stores both presented stimuli and stimulus
>       templates, the latter in case the same stimulus is presented multiple times,
>       or is pulled from an external stimulus library. Stimuli are here
>       defined as any signal that is pushed into the system as part of the experiment
>       (eg, sound, video, voltage, etc). Many different experiments can use the same
>       stimuli, and stimuli can be re-used during an experiment. The stimulus group
>       is organized so that one version of template stimuli can be stored and these
>       be used multiple times. These templates can exist in the present file or can
>       be linked to a remote library file.'

https://github.com/NeurodataWithoutBorders/nwb-schema/blob/77b424a336a34038b133f3858aff822f01057665/core/nwb.file.yaml#L115

My reading of this schema would be that shocks could be added as `stimulus`, but I don't have a strong opinion about this. Also don't have an example from before unfortunately.